src/App.js

```diff
@@ -89,15 +88,45 @@ class App extends React.Component {
             delete updateDataByStatuses[status];            
         }
         
-        this.setState({            
-            dataByStatuses: updateDataByStatuses
+        this.setState({       
+            dataCard: {
+                dataByStatuses: updateDataByStatuses,
+                statuses
+            },
+            modalWindow: this.state.modalWindow            
         });
     }
 
+    handleModalInfo = (card) => {
+        document.body.style.overflow = "hidden";
+        
+        this.setState({
+            dataCard: this.state.dataCard,
+            modalWindow: {
+                visible: true,
+                card
+            }
+        })        
+    }
+
+    handleCloseModal = () => {
+        document.body.style.overflow = "visible";
+
+        this.setState({
+            dataCard: this.state.dataCard,
+            modalWindow: {
+                visible: false,
+                card: this.state.modalWindow.card
```

Comment:
please save card id only

Reply:
but it’s easier to store a link to a card so that you can get the information you need later.

if you store only id, then you will need to do unnecessary actions associated with filtering cards by id

Reply:
@DmitryApan you are wrong, you can pass status and id from card on open

Reply:
> @DmitryApan you are wrong, you can pass status and id from card on open

The card component receives the card object from the section component. I can’t understand why I need to pass id if I can pass a link to the map object that I want to display in a modal window?

if I pass id, then I will need to write an additional function to find a card with a specific id. What for? Will it be necessary for future functionality?

Reply:
@DmitryApan in the next feature you will load card itself (title, description, status, comments etc) you won't have all of these data in props, you will load them from another endpoint and you will need only id for this
it means you can leave all things as is but you will refactor all of this later